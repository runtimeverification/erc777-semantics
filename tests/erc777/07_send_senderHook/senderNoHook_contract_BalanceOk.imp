test.preamble();

// send() to other address,  operatorSend() through default operator, operatorSend() from caller.
// contract holder address with no hook, sending sufficient balance

test.makeCaller(10);
test.send(3, 10, "");
test.operatorSend(10, 3, 10, "", ""); //from caller

test.makeCaller(8);
test.operatorSend(10, 3, 10, "", ""); //through default operator
