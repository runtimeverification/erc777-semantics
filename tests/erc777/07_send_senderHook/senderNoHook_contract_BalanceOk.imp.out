Changing the caller to 10
10 is sending 10 to 3 (data:)
Balance of 10 is 90
Balance of 3 is 40
10 is sending 10 from 10 to 3 (data:""),  (operatorData:"")
Balance of 10 is 80
Balance of 3 is 50
Changing the caller to 8
8 is sending 10 from 10 to 3 (data:""),  (operatorData:"")
Balance of 10 is 70
Balance of 3 is 60
